TikTokApi/tiktok.py

```diff
@@ -160,14 +169,22 @@ def getData(self, **kwargs) -> dict:
             proxies=self.__format_proxy(proxy),
         )
         try:
+            json = r.json()
+            if "type" in json and json["type"] == "verify" :
+                logging.error("Tiktok wants to display a catcha. Response is:\n" + r.text)
+                raise TikTokCaptchError()
             return r.json()
-        except Exception as e:
-            logging.error(e)
-            logging.error(
-                "Converting response to JSON failed response is below (probably empty)"
-            )
-            logging.info(r.text)
-            raise Exception("Invalid Response")
+        except JSONDecodeError as e:
+            text = r.text
+            logging.error("TikTok response: " + text)
+            if len(text) == 0 :
+                raise Exception("Empty response from Tiktok to " + url) from None
```

Comment:
I would define specific error classes for these.  That would allow consumers of this API to handle specific errors in idiomatic ways (i.e. with `except` clauses targeting specific exception classes).  Using generic `Exception` forces consumers to inspect the contents of the exception object to determine how to handle it, which can be a code smell.

I'd also recommend putting all exception classes in a top-level `exceptions` module for easy importing.

Reply:
Not sure for this one to because it should no occur. Empty response is usually due error in the request, bad signature, api change, missing param, etc...

Reply:
If it should not occur, then why handle it at all?  It's better to allow an unexpected exception to propagate up the entire frame stack and terminate the program, because otherwise errors (and the bugs they may indicate) could be discarded silently.